app/views/my/solutions/_show_legacy.html.haml

```diff
@@ -11,18 +11,8 @@
       What's next?
 
     .next-option
-      -if @user_track.mentoring_slots_remaining?
-        %p
-          %strong Request mentor feedback.
-          This solution has been imported from an old version of the website.
-          You may request mentoring to unlock its extra exercises.
-        =link_to "Request mentor feedback", request_mentoring_my_solution_path(@solution), method: :patch, class: 'pure-button'
-      -else
-        %p
-          %strong Request mentor feedback (disabled).
-          This solution has been imported from Practice Mode.
-          Once your existing solutions have been mentored you will be able to request mentoring for this solution.
-        =link_to "Request mentor feedback", "#", class: 'pure-button disabled', disabled: true
+      = render "my/solutions/mentor_feedback_request/legacy",
+        mentor_request: MentorFeedbackRequest.new(@solution)
```

Comment:
I think the `app/models/mentor_feedback_request.rb` could then be moved into this same helper as a subclass, so something like this:

```ruby
module SolutionMentorFeedbackRequestSection
  class MentorFeedbackRequest
  end

  def render_solution_mentor_feedback_request_section(scenario, solution)
    mentor_request = MentorFeedbackRequest.new(solution)
    render "my/solutions/mentor_feedback_request/independent/#{mentor_request.status}",
  mentor_request: mentor_request
  end
end
```

You might even be able to extract the method from `MentorFeedbackRequest` entirely and just embed it into the helper method

Reply:
I've gone with the helper method route, but decided to keep `MentorFeedbackRequest` as a first class object because I was able to use it in `RequestMentoringOnSolution`. This is a nice way to keep our logic in just one place.